floodwatch/src/js/components/Profile.js

```diff
@@ -0,0 +1,198 @@
+        </Col>
+      </Row>
+    )   
+  }
+
+}
+
+export class DemographicContainer extends Component {
+  state: StateType
+
+  constructor() {
+    super();
+    this.state = {
+      toPick: ['birth_year', 'twofishes_id', 'demographic_ids'],
+      curStatus: ''
+    }
+  }
+
+  async componentDidMount() {
+    const UserData = await FWApiClient.get().getCurrentPerson()
+    let filteredUserData = _.pick(UserData, this.state.toPick)
+
+    this.setState({
+      userData: filteredUserData
+    })
+  }
+
+  handleClick(checked: boolean, event: any): void {
+    let demo = event.target.textContent;
+    let demo_id = _.find(DemographicKeys.demographic_keys, (o: DemographicDictionary) => {
+      return o.name == demo
+    })
+
+    if (event.target.name != 'age' && event.target.name != 'country_code') {
+      console.log(checked)
+      if (checked) {
+        this.addToDemographicIds(demo_id.id)
+      } else {
+        this.removeFromDemographicIds(demo_id.id)
+      }
+    }
+  }
+
+  async updateUserInfo() {
+    try {
+      const reply = await FWApiClient.get().updatePersonDemographics(this.state.userData)  
+      this.setState({curStatus: 'success'})
+    } catch (e) {
+      console.error(e)
+      this.setState({curStatus: 'error'})
+    }
+  }
+
+
+
+  addToDemographicIds(id: number): void {
+    let userData = _.cloneDeep(this.state.userData);
+    userData.demographic_ids.push(id)
+    this.setState({
+      userData: userData,
+      curStatus: (this.state.curStatus != 'error') ? '' : 'error'
+    })
+  }
+
+  updateYear(event: any): void {
+    let userData = _.cloneDeep(this.state.userData);
+    userData.birth_year = parseInt(event.target.value);
+    this.setState({
+      userData: userData,
+      curStatus: (this.state.curStatus != 'error') ? '' : 'error'
+    })
+  }
+
+  removeFromDemographicIds(id: number): void {
+    let userData = _.cloneDeep(this.state.userData)
+    userData.demographic_ids = _.filter(userData.demographic_ids, (o) => {
+      return o != id
+    })
+    this.setState({
+      userData: userData,
+      curStatus: (this.state.curStatus != 'error') ? '' : 'error'
+    })
+  }
+
+  render() {
+    let elems = Filters.filters.map((filter) => {
+      return <DemographicOption updateYear={this.updateYear.bind(this)} 
+                                handleClick={this.handleClick.bind(this)} 
+                                userData={this.state.userData} filter={filter}/>
+    })
+
+    const displayGroup = (this.state.curStatus == '') ? 'none' : 'block'
+
+    return (
+      <Row style={{border:'1px solid #e9e9e9', borderRadius:'10px', padding:'20px'}}> 
+        <Col xs={12}>
+        {elems}
+        </Col>
+        <Col xs={12}>
+        <Button onClick={this.updateUserInfo.bind(this)} style={{backgroundColor:'lightblue', width:'100%'}}>Submit</Button>
+        <ListGroup style={{display: displayGroup, textAlign:'center'}}>
+          <ListGroupItem bsStyle="success" 
```

Comment:
This might be cleaner as:

```jsx
{ this.state.curStatus == 'success' &&
  <ListGroupItem bsStyle="success">
    Successfully saved changes!
  </ListGroupItem> }
{ this.state.curStatus == 'error' &&
  <ListGroupItem bsStyle="danger">
    Error while trying to save changes. Please check your connection.
  </ListGroupItem> }
```